Feat/guard routes based on roles

jphinning opened the pull request:
- Separated routes between private, public, freelancer, and client roles
- Changed folder structure of the pages folder
- Made new route guards, freelancers won't be able to access client routes and vice-versa

- Changed the navbar to include the sign-out button when the drawer is rendered for mobile screen resolutions:
-- Client:
![image](https://user-images.githubusercontent.com/61670871/188044752-118ed3c5-ed78-43d2-886b-09be46915f0b.png)
-- Freelancer:
![image](https://user-images.githubusercontent.com/61670871/188044830-dcba99a6-872f-4589-8a6f-78a00e6e811a.png)

jphinning committed:
feat: Protect freelancer and client roles

jphinning committed:
feat: Navigate to different routes depending on role

jphinning committed:
NavBar and drawer updated protected depending on role

jphinning committed:
fix: Role saving previous user state fixed

jphinning committed:
fix: Google unwanted signIn fixed

jphinning committed:
fix: App flow now works as intended

jphinning committed:
Merge branch 'dev' into feat/guard-routes-based-on-roles

jphinning committed:
fix: Remove unnecessary comment

jphinning committed:
fix: Add route const instead of string

jphinning committed:
fix: Consistent styles for both job details and job list

jphinning committed:
fix: Consistent styles for both job details and job list

jphinning committed:
Merge branch 'dev' into feat/guard-routes-based-on-roles

xx-TOOL-xx commented on src/components/navbar/Navbar.tsx:
```diff
@@ -90,7 +51,7 @@ function Navbar() {
   return (
     <NavbarStyles>
-      <NavLink path={role === "freelancer" ? HOME_PAGE_ROUTE : CLIENT_HOME}>
+      <NavLink path={role === "freelancer" ? SEARCH_WORK_ROUTE : CLIENT_HOME}>
```
don't compare with string. Move it to const value

jphinning replied:
Modified the NavLinksPerRole object, now every item rendered in the navBar is based on the role stored in redux persist according to that object

xx-TOOL-xx commented on src/pages/Auth/Welcome/index.tsx:
```diff
@@ -1,21 +1,25 @@
-import { RootState } from "store/store";
-import { PROFILE_ROUTE } from "utils/routeConsts";
+import { CLIENT_PROFILE, PROFILE_ROUTE } from "utils/routeConsts";
 import S from "./style";
```
absolute path

jphinning replied:
Done 👍🏻

jphinning committed:
Conditional render parameterized

The navItens object was modified to be able not generate any simple string comparisons in the NavBar component such as {role === freelancer?}

lyudad commented on src/components/UI/JobItem/JobItem.tsx:
```diff
@@ -6,17 +6,22 @@ import { useTranslation } from "react-i18next";
 
 interface Props {
```
better to put `I` at the name of the interface so its will be quicker to understand that this is interface - IProps instead of Props

jphinning replied:
Ok

lyudad commented on src/pages/Auth/CompleteRegistration/CompleteRegistration.tsx:
```diff
@@ -1,70 +1,64 @@
-      id: sub,
-      role,
+    await updateRole({
+      id,
```
why id? if you have a token already? if you have token you know who is the user

jphinning replied:
Yes, but the backend has an endpoint that expects an ID in order to update the user. 
Since we can't store the ID and the token has the information we need, I'm using the hook useJwtDecoder() to decode and return the ID so I can pass it to the update role request.

Are you suggesting, for example, that I change the backend endpoints to instead of /settings-info/:id to /settings-info/self?
And then, since we're using passport, simply use the value we have stored in req.user.id to make queries to the db?

lyudad commented on src/hooks/useJwtDecoder.tsx:
```diff
@@ -4,9 +4,13 @@ import jwtDecode, { JwtPayload } from "jwt-decode";
+  return { sub: "0", exp: "0" };
 };
 
 export default useJwtDecoder;
```
why this file tsx? it does not return component

jphinning replied:
Fixed

lyudad commented on src/hooks/useJwtDecoder.tsx:
```diff
@@ -4,9 +4,13 @@ import jwtDecode, { JwtPayload } from "jwt-decode";
+  }
+
+  return { sub: "0", exp: "0" };
 };
```
I dont like this function, I dont see reason to use it

jphinning replied:
This one returns the id and expiration date stored in the JWT token. Besides the id, I believe it will be useful in the future to implement auto logout after token expiration.

jphinning committed:
fix: types and file names fixed

jphinning committed:
fix: Job list query fixed

jphinning committed:
fix: Update and getOwn operations won't need the user id anymore
